src/sortStudents.ts

```diff
@@ -1,16 +1,70 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desk';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+):Student[] {
+  const studentArr: Student[] = [...students];
+
+  function getGrades(arr: number[]): number {
+    return arr.reduce((a: number, b: number) => a + b) / arr.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.name.localeCompare(b.name)
+        : b.name.localeCompare(a.name)));
+      break;
+
+    case SortType.Surname:
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.surname.localeCompare(b.surname)
+        : b.surname.localeCompare(a.surname)));
+      break;
```

Comment:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      studentArr.sort((a: Student, b: Student) => (order === 'asc'
        ? a[sortBy].localeCompare(b[sortBy])
        : b[sortBy].localeCompare(a[sortBy])));
      break;
```
this to cases with same logic